src/address.js

```diff
@@ -10,11 +10,27 @@ const typeforce = require('typeforce');
 function fromBase58Check(address) {
   const payload = bs58check.decode(address);
   // TODO: 4.0.0, move to "toOutputScript"
-  if (payload.length < 21) throw new TypeError(address + ' is too short');
-  if (payload.length > 21) throw new TypeError(address + ' is too long');
+  if (payload.length < 21)
+    throw new TypeError(`${address} is too short(${payload.length})`);
+  if (payload.length > 54)
+    throw new TypeError(`${address} is too long(${payload.length})`);
   const version = payload.readUInt8(0);
-  const hash = payload.slice(1);
-  return { version, hash };
+  if (payload.length > 21) {
+    // Colored
+    const colorId = payload.slice(1, 34);
+    const hash = payload.slice(34);
+    if (hash.length !== 20) {
+      throw new TypeError(`Invalid hash(${hash})`);
+    }
+    return { version, colorId, hash };
+  } else {
+    // Uncolored
+    const hash = payload.slice(1);
+    if (hash.length !== 20) {
+      throw new TypeError(`Invalid hash(${hash})`);
```

Comment:
In the first check, since `21 <= payload < 54` and `payload.length > 21`, the payload in this branch is always 21, so this if check is not really necessary, is it?

Reply:
Add commit (06be2599ad479c56719caf03ba00a82169d122e4)